### PR TITLE
Update object-and-collection-initializers.md

### PR DESCRIPTION
**Pull request by bigboybamo**
Added section for class type object Initialization

## Summary

Fix for  #40369

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/programming-guide/classes-and-structs/object-and-collection-initializers.md](https://github.com/dotnet/docs/blob/9bfde212abb72f12da3d76d485c78b44d8027815/docs/csharp/programming-guide/classes-and-structs/object-and-collection-initializers.md) | [Object and Collection Initializers (C# Programming Guide)](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers?branch=pr-en-us-40466) |

### CODE REVIEW

**Comment by BillWagner** [comment]
Hi @bigboybamo 

I think this is a great start. I've got some initial thoughts on how to make the explanations more clear for more readers.

1. In your sample, consider changing the class names to have meaning in the context of the explanation. Something like `ContainerClass`, `EmbeddedClassType` or such.
2. Add a default constructor for the embedded class type, and write the values of the properties in that constructor. That way, readers can see the order of creating the embedded object, setting its initial values, and calling the object initializers.

Also, a few suggestions to match our style:

1. The sample should be a separate program and included in using the `:::` syntax. That way, the code is run through our CI system and we make sure it's always valid.
2. The explanation should be in text, not in comments (or both). The reason is that comments aren't translated into all the human languages we support, but the text is. It helps non-English speakers read the explanations in their native language.

I'll watch for updates and give a more detailed review when you're ready.

**Comment by bigboybamo** [comment]
Hi @BillWagner  I made a couple of commits. Please review and let me know what you think. 
Thanks

File: samples/snippets/csharp/programming-guide/classes-and-structs/object-collection-initializers/HowToClassTypedInitializer.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            public string BS { get; set; }
+
+            public override string ToString() => $"{BI}|{BB}|{BS}";
+
+            public EmbeddedClassTypeB()
+            {
+                BI = 23;
+                BB = false;
+                BS = "BBBabc";
+            }
```
Comment (BillWagner) [inline samples/snippets/csharp/programming-guide/classes-and-structs/object-collection-initializers/HowToClassTypedInitializer.cs:42]: Here's one of the locations where I'd add information about the properties to show the order of initialization:

```suggestion
            public EmbeddedClassTypeB()
            {
               Console.WriteLine($"Entering EmbeddedClassTypeB constructor. Values are: {this}");
                BI = 23;
                BB = false;
                BS = "BBBabc";
                Console.WriteLine($"Exiting EmbeddedClassTypeB constructor. Values are: {this})");
            }
```

Then, continue that process with the other constructors, and in the `Main` method after the initialization happens.

**Comment by bigboybamo** [comment]
Hi @BillWagner I think I have got what you meant, I have updated the PR again.
Thanks

**Review by BillWagner — APPROVED** [review]
Thanks again @bigboybamo 

This looks great, and I'll :shipit: now.

**Pull request merged**